x/oracle/keeper/grpc_query.go

```diff
@@ -262,15 +263,17 @@ func (q querier) Medians(
 
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
+	if req.NumStamps > q.MaximumMedianStamps(ctx) {
+		return nil, status.Error(codes.OutOfRange, fmt.Sprintf("numStamps must be less than %d", q.MaximumMedianStamps(ctx)))
+	}
+
 	var medians sdk.DecCoins
 
 	if len(req.Denom) > 0 {
-		// TODO: Add numStamps param to request to enable responding with a variable amount
-		// of last medians rather than just returning the most recent stamp
-		medianList := q.HistoricMedians(ctx, req.Denom, 1)
+		medianList := q.HistoricMedians(ctx, req.Denom, req.NumStamps)
 
-		if len(medianList) != 0 {
-			medians = medians.Add(sdk.NewDecCoinFromDec(req.Denom, medianList[0]))
+		for _, median := range medianList {
+			medians = medians.Add(sdk.NewDecCoinFromDec(req.Denom, median))
```

Comment:
Don't use `Add`. Create the slice with the expected size and just set the values. It's faster. We need to be fast in this query.